infra/launch_on_ray.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+# Similar to launch.py, but this instead launches on a Ray cluster configured with auto-scaling TPUs
+
+import argparse
+import getpass
+import os
+import tempfile
+import time
+from pathlib import Path
+
+import draccus
+from ray.dashboard.modules.job.common import JobStatus
+from ray.dashboard.modules.job.sdk import JobSubmissionClient
+
+import levanter.infra.cli_helpers as cli
+import levanter.infra.docker as docker
+
+
+def main():
```

Comment:
If we were launching jobs from the Marin Executor, we just need Ray remote function to call into...but this doesn't seem like it?

Reply:
This is for non-Marin use cases or for jobs that aren't immediately part of the pipeline.

Reply:
(I'll add stuff into marin directly for marin. This is the core functionality plus the CLI we need.)